pxc/jenkins/pxc56-pipeline.groovy

```diff
@@ -0,0 +1,261 @@
+            choices: '/usr/bin/cmake',
+            description: 'path to cmake binary',
+            name: 'JOB_CMAKE')
+        choice(
+            choices: 'RelWithDebInfo\nDebug',
+            description: 'Type of build to produce',
+            name: 'CMAKE_BUILD_TYPE')
+        string(
+            defaultValue: '',
+            description: 'cmake options',
+            name: 'CMAKE_OPTS')
+        string(
+            defaultValue: '',
+            description: 'make options, like VERBOSE=1',
+            name: 'MAKE_OPTS')
+        choice(
+            choices: 'yes\nno',
+            description: 'Run mysql-test-run.pl',
+            name: 'DEFAULT_TESTING')
+        string(
+            defaultValue: '--unit-tests-report --suite=galera,galera_3nodes,galera_sr,galera_3nodes_sr,sys_vars',
+            description: 'mysql-test-run.pl options, for options like: --big-test --only-big-test --nounit-tests --unit-tests-report',
+            name: 'MTR_ARGS')
+        string(
+            defaultValue: '1',
+            description: 'Run each test N number of times, --repeat=N',
+            name: 'MTR_REPEAT')
+    }
+    agent {
+        label 'micro-amazon'
+    }
+    options {
+        skipDefaultCheckout()
+        skipStagesAfterUnstable()
+        timeout(time: 6, unit: 'DAYS')
+        buildDiscarder(logRotator(numToKeepStr: '200', artifactNumToKeepStr: '200'))
+    }
+    stages {
+        stage('Prepare') {
+            steps {
+                script {
+                    currentBuild.displayName = "${BUILD_NUMBER} ${CMAKE_BUILD_TYPE}/${DOCKER_OS}"
+                }
+
+                sh 'echo Prepare: \$(date -u "+%s")'
+                echo 'Checking PXC branch version'
+                sh '''
+                    MY_BRANCH_BASE_MAJOR=5
+                    MY_BRANCH_BASE_MINOR=6
+                    RAW_VERSION_LINK=$(echo ${GIT_REPO%.git} | sed -e "s:github.com:raw.githubusercontent.com:g")
+                    wget ${RAW_VERSION_LINK}/${BRANCH}/VERSION -O ${WORKSPACE}/VERSION-${BUILD_NUMBER}
+                    source ${WORKSPACE}/VERSION-${BUILD_NUMBER}
+                    if [[ ${MYSQL_VERSION_MAJOR} -lt ${MY_BRANCH_BASE_MAJOR} ]] ; then
+                        echo "Are you trying to build wrong branch?"
+                        echo "You are trying to build ${MYSQL_VERSION_MAJOR}.${MYSQL_VERSION_MINOR} instead of ${MY_BRANCH_BASE_MAJOR}.${MY_BRANCH_BASE_MINOR}!"
+                        rm -f ${WORKSPACE}/VERSION-${BUILD_NUMBER}
+                        exit 1
+                    fi
+                    rm -f ${WORKSPACE}/VERSION-${BUILD_NUMBER}
+                '''
+            }
+        }
+        stage('Check out and Build Galera/PXB') {
+            parallel {
+                stage('Build Galera library') {
+                    agent { label 'docker' }
+                    steps {
+                        git branch: 'PXC-3285-add-pipeline-jobs', url: 'https://github.com/Percona-Lab/jenkins-pipelines'
```

Comment:
set 'master' branch  for https://github.com/Percona-Lab/jenkins-pipeline

Reply:
done